### DISCUSSION TITLE
Nested Control DataContext

### DISCUSSION THREAD

**Discussion by jaymarvels** [AvaloniaUI/Avalonia#15661] [Q&A]
My understanding from this diagram:

![image](https://github.com/AvaloniaUI/Avalonia/assets/7057631/44cfd7af-e04f-4c5c-b8f9-ae9b564d6053)

Is that a control inside a control can get use to context "higher up" - "... control defined in a window can use the data context of the window; or (as above) a control in a control in a window can use the window's data context..."

So I have my MainWindow and inside this I have my MainView, inside my MainView I have other controls (views), my MainView has its own MainViewModel and in my App.axmal.cs I set up my DI for them:

```
    public override void OnFrameworkInitializationCompleted() 
    {
        // Register all the services needed for the application to run
        var collection = new ServiceCollection();
        collection.AddTransient<MainViewModel>();
        collection.AddTransient<CitiesViewModel>();
        collection.AddSingleton<IDialogService, DialogService>();
        collection.ConfigureFrameworkServices();

        // Creates a ServiceProvider containing services from the provided IServiceCollection
        var services = collection.BuildServiceProvider();

        var vm = services.GetRequiredService<MainViewModel>();

        if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop) 
        {
            desktop.MainWindow = new MainWindow() 
            {
                DataContext = vm
            };
        }
        else if (ApplicationLifetime is ISingleViewApplicationLifetime singleViewPlatform) 
        {
            singleViewPlatform.MainView = new MainView() 
            {
                DataContext = vm
            };
        }

        base.OnFrameworkInitializationCompleted();
    }
```

In my MainViewModel I have an `ObservableCollection` which I can display in a DataGrid on my MainView without issue.

One of my controls - CityPage -  inside the MainView, wants to use this ObservableCollection in its own View, so I attempt something like this:

```
    <DataGrid Margin="20" ItemsSource="{Binding Items, Mode=TwoWay}"
              AutoGenerateColumns="True" IsReadOnly="False"
              GridLinesVisibility="All"
              BorderThickness="1" BorderBrush="Gray">
  </DataGrid>
```
It complains about the xData type so I have to add in `x:DataType="viewModels:MainViewModel"` and at this point it populates the DataGrid using the ObservableCollection from the "parent" control.

One of the issues I am seeing though is if I try and make a change to a value in the DataGrid thats on the CityPage this doesn't get reflected/updated on the bound ObservableCollection that is displayed on the MainView. Doing it the other way around (so changing on the MainView) does automatically change the value of the child control (CityPage).

In my MainViewModel I am doing:

```
    public ObservableCollection<CityModel> Items
    {
        get => _items;
        set => this.RaiseAndSetIfChanged(ref _items, value);
    }
```

So it should fire off the RaiseAndSetIfChanged.

Am I missing something in the way datacontext is used and passed around? As you can see I have tried "TwoWay" mode too without much luck.

**Comment by timunie** [comment]
Your items need to raise property changed, not the collection itself
